mesonbuild/interpreter/mesonmain.py

```diff
@@ -290,6 +291,12 @@ def _can_run_host_binaries_impl(self) -> bool:
             self.build.environment.exe_wrapper is None
         )
 
+    @noPosargs
+    @noKwargs
+    @FeatureNew('meson.implementation', '1.3.0')
+    def implementation_method(self, args: T.List['TYPE_var'], kwargs: 'TYPE_kwargs') -> str:
```

Comment:
Do you need to quote `TYPE_var` and `TYPE_kwargs` here?

Reply:
Maybe even `T.Tuple[void]` and `T.Dict[void]`? Dunno how to annotate empty tuple/dict, @dcbaker maybe has idea how to properly type functions that uses `@noPosargs` and `@noKwargs`.

Reply:
It does not need quoting, the quotes are harmless other than taking up an entire two additional bytes of disk space. Quite some time ago I upgraded the entire codebase to consistently use `from __future__ import annotations`, which causes `args: T.List['TYPE_var']` to be interpreted as `args: 'T.List[TYPE_var]'` anyway. No manual quoting needed.

Reply:
that is typed correctly to fulfill the contract.